### ISSUE TITLE
invalid call to xmlFree() lead linux program to crash

### ISSUE THREAD

**Issue by cberge** [OData/odatacpp-server#16]
In source file  odatacpp-server/src/common/xmlhelpers.cpp :
in function void xml_writer::write_start_element_with_prefix() 
The code dedicated to Linux is the following:
# else

```
    xmlChar* valueXml = (xmlChar*) ::odata::utility::conversions::to_utf8string(elementName).c_str();
    xmlTextWriterStartElement(m_writer, valueXml);
    xmlFree(valueXml);
```
# endif

I'm not sure that the call to xmlFree() is required as valueXml wasn't allocated thought xml stuff.
Anyway, this call to xmlFree() make the program crash on linux...

Can someone check what I say is correct?
Thanks

Christian

**Comment by lewischeng-ms** [comment]
@Bernat65 Thanks for reporting this issue. Given that `elementName` is not necessarily allocated using `xmlMalloc`, the use of `xmlFree` to release the buffer is incorrect. `elementName` should have its own life cycle management for its internal buffer. Would you please send us a pull request to fix this?

**Comment by cberge** [comment]
Thanks Lewis,
I’m not a GitHub expert but I’m pleased if I can help ☺
I have some tasks to finish first, but I’ll will do that as soon as possible (maybe next week)
Thanks again,
Christian

PS: It sounds like the linux code wasn’t fully tested and so, I found a few errors. I’ll will process them one by one.

**Comment by lewischeng-ms** [comment]
Resolved by PR #18. Thanks for your contribution!
